youtube_dl/extractor/ellentube.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    clean_html,
+    int_or_none,
+    urljoin,
+)
+
+
+class EllenTubeBaseIE(InfoExtractor):
+    API_URL = 'https://api-prod.ellentube.com/'
+
+    def _extract_video_from_json(self, data, video_id, display_id=None):
+        title = data['title']
+        description = data.get('description')
+        publish_time = int_or_none(data.get('publishTime'))
+        thumbnail = data.get('thumbnail')
+
+        formats = []
+        duration = None
+        for entry in data.get('media'):
+            if entry.get('id') == 'm3u8':
+                formats = self._extract_m3u8_formats(
+                    entry.get('url'), video_id, 'mp4', entry_protocol='m3u8_native', m3u8_id='hls')
+                duration = int_or_none(entry.get('duration'))
+                break
+        self._sort_formats(formats)
+        return {
+            'id': video_id,
+            'title': title,
+            'description': description,
+            'display_id': display_id,
+            'duration': duration,
+            'thumbnail': thumbnail,
+            'timestamp': publish_time,
+            'formats': formats,
+        }
+
+    def _extract_playlist_entries_from_json(self, data, display_id):
+        return [self._extract_video_from_json(elem, elem['id'])
+                for elem in data if elem.get('type') == 'VIDEO']
+
+    def _extract_from_video_id(self, video_id, display_id=None):
+        api_data = self._download_json(
+            urljoin(self.API_URL, 'ellenapi/api/item/%s' % video_id), video_id)
+        return self._extract_video_from_json(api_data, video_id, display_id)
+
+    def _extract_playlist(self, url, display_id, extract_description=True):
+        webpage = self._download_webpage(url, display_id)
+        playlist_data = self._html_search_regex(
+            r'<div\s+data-component\s*=\s*"Details"(.+)</div>', webpage, 'playlist data')
+        playlist_title = self._search_regex(
+            r'"title"\s*:\s*"(.+?)"', playlist_data, 'playlist title')
+        playlist_description = clean_html(self._search_regex(
+            r'"description"\s*:\s*"(.+?)"', playlist_data, 'playlist description',
+            fatal=False)) if extract_description else None
+        api_search = self._search_regex(
+            r'"filter"\s*:\s*"(.+?)"', playlist_data, 'playlist api request')
+        api_data = self._download_json(
+            urljoin(self.API_URL, 'ellenapi/api/feed/?%s' % api_search), display_id)
+        return self.playlist_result(
+            self._extract_playlist_entries_from_json(api_data, display_id),
+            display_id, playlist_title, playlist_description)
```

Comment:
Currently, video identifying is completely messed.
This is how extractors should be organized:
1. `EllenTubeIE` accepting `https://api-prod.ellentube.com/ellenapi/api/item/uuid` and `ellentube:uuid`.
2. All other extractors must delegate to it so download archive is always written as `ellentube`.